node/src/components/consensus/highway_core/state/params.rs

```diff
@@ -89,6 +92,11 @@ impl Params {
         self.end_height
     }
 
+    /// Returns the start timestamp of the era.
+    pub(crate) fn start_timestamp(&self) -> Timestamp {
+        self.start_timestamp
+    }
```

Comment:
Is that duplicated now? I think the `ActiveValidator` receives this as an explicit argument, too?

If we add that to `Params`, let's:
* make votes invalid if they are before the start time, and
* make the `ActiveValidator` take the start value from the `Params` instead of as a redundant parameter.

Reply:
True, it does, but at a slightly different point. I don't think we can reuse it. I could make it not be a part of `Params`, but an additional parameter to the `HighwayProtocol`'s constructor, though - whichever you prefer.

Reply:
Why can't we reuse it? Doesn't `ActiveValidator::new` have access to `Params`?

Reply:
Not currently, but I think I should be able to change it. I meant it the other way round - that it would be hard to reuse the timestamp `ActiveValidator` receives at the moment, as it receives it via a call to `activate_validator`, which is something `HighwayProtocol` has no control over.

Reply:
Exactly: Using `Params` in `ActiveValidator` sounds like the right thing to do. :+1:

Reply:
Actually, after another look, I'm not sure if there even is duplication. What `ActiveValidator` receives is not necessarily the same as the starting timestamp - what it gets is `timestamp.max(start_time)`, where `timestamp` is the current timestamp, and `start_time` is what gets stored in `Params`.
So while we could pass `Params` to `ActiveValidator`, too, I'm actually thinking that maybe we're doing it right as it is.

Reply:
Ah, I see, this is so that `ActiveValidator` can set the timer accordingly.
But then I'd at least always pass in the curren timestamp, and do the `max` inside the method, so the caller doesn't have to care about that.

Reply:
Do you mean inside `activate_validator`, or inside `ActiveValidator::new`? I'd prefer the former, as the latter causes trouble with testing (all tests of `ActiveValidator` would have to instantiate `Params` now :grimacing: ).

Reply:
I was thinking inside `ActiveValidator::new`… but that's a good point, of course. Let's not do that in this PR then. :+1: 
Inside `activate_validator` would still be an improvement compared to having it in the era supervisor.